configure.ac

```
@@ -1658,220 +1659,45 @@ AC_MSG_NOTICE([$KRB5_FLAVOUR Kerberos library support: ${with_krb5:=no} ${LIB_KR
+  dnl On MinGW OpenLDAP is not available, so LDAP helpers can be linked
+  dnl only with Windows LDAP libraries using -lwldap32
+  AS_IF([test "$squid_host_os" = "mingw"],[
+    LIBLDAP_LIBS="-lwldap32"
   ],[
-    AC_MSG_RESULT(cross-compiler cant tell)
-  ])
-
-  AC_MSG_CHECKING([for LDAP_REBINDPROC_CALLBACK])
-  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
-#   if HAVE_LDAP_H
-#   include <ldap.h>
-#   elif HAVE_MOZLDAP_LDAP_H
-#   include <mozldap/ldap.h>
-#   endif
-  ]],[[
-    LDAP_REBINDPROC_CALLBACK ldap_rebind;
-  ]])],[
-    AC_DEFINE(HAVE_LDAP_REBINDPROC_CALLBACK,1,[Define to 1 if you have LDAP_REBINDPROC_CALLBACK])
-    AC_MSG_RESULT(yes)
-  ],[
-     AC_MSG_RESULT(no)
-  ])
-
-  AC_MSG_CHECKING([for LDAP_REBIND_PROC])
-  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
-#   if HAVE_LDAP_H
-#   include <ldap.h>
-#   elif HAVE_MOZLDAP_LDAP_H
-#   include <mozldap/ldap.h>
-#   endif
-  ]],[[
-    LDAP_REBIND_PROC ldap_rebind;
-  ]])],[
-    AC_DEFINE(HAVE_LDAP_REBIND_PROC,1,[Define to 1 if you have LDAP_REBIND_PROC])
-    AC_MSG_RESULT(yes)
-  ],[
-    AC_MSG_RESULT(no)
-  ])
-
-  AC_MSG_CHECKING([for LDAP_REBIND_FUNCTION])
-  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
-#   define LDAP_REFERRALS
-#   if HAVE_LDAP_H
-#   include <ldap.h>
-#   elif HAVE_MOZLDAP_LDAP_H
-#   include <mozldap/ldap.h>
-#   endif
-  ]],[[
-    LDAP_REBIND_FUNCTION ldap_rebind;
-  ]])],[
-    AC_DEFINE(HAVE_LDAP_REBIND_FUNCTION,1,[Define to 1 if you have LDAP_REBIND_FUNCTION])
-    AC_MSG_RESULT(yes)
-  ],[
-    AC_MSG_RESULT(no)
+    SQUID_STATE_SAVE(squid_ldap_state)
+    LIBS="$LIBLDAP_PATH $LIBPTHREADS $LIBS"
+    PKG_CHECK_MODULES([LIBLDAP],[ldap],[],[
+      AC_CHECK_LIB(lber, ber_init, [LIBLBER="-llber"])
+      AC_CHECK_LIB(ldap, ldap_init, [LIBLDAP_LIBS="-lldap $LIBLBER"])
+      dnl if no ldap lib found check for mozilla version
+      AS_IF([test "x$ac_cv_lib_ldap_ldap_init" != "xyes"],[
+        SQUID_STATE_SAVE(squid_ldap_mozilla)
+        LIBS="$LIBLDAP_PATH $LIBPTHREADS"
+        AC_CHECK_LIB(ldap60, ldap_init, [LIBLDAP_LIBS="-lldap60 $LIBLBER"])
+        LIBS="$LIBLDAP_PATH $LIBLDAP_LIBS $LIBPTHREADS"
+        AC_CHECK_LIB(prldap60, prldap_init, [LIBLDAP_LIBS="-lprldap60 $LIBLDAP_LIBS"])
+        LIBS="$LIBLDAP_PATH $LIBLDAP_LIBS $LIBPTHREADS"
+        AC_CHECK_LIB(ssldap60, ldapssl_init, [LIBLDAP_LIBS="-lssldap60 $LIBLDAP_LIBS"])
+        SQUID_STATE_ROLLBACK(squid_ldap_mozilla)
+      ])
+    ])
+    AC_CHECK_HEADERS(ldap.h lber.h)
+    AC_CHECK_HEADERS(mozldap/ldap.h)
+    SQUID_CHECK_LDAP_API
   ])
 
-  AC_MSG_CHECKING([for LDAP_SCOPE_DEFAULT])
-  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
-#   if HAVE_LDAP_H
-#   include <ldap.h>
-#   elif HAVE_MOZLDAP_LDAP_H
-#   include <mozldap/ldap.h>
-#   endif
-  ]],[[
-    int i=LDAP_SCOPE_DEFAULT;
-  ]])],[
-    AC_DEFINE(HAVE_LDAP_SCOPE_DEFAULT,1,[Define to 1 if you have LDAP_SCOPE_DEFAULT])
-    AC_MSG_RESULT(yes)
+  AS_IF([test "x$LIBLDAP_LIBS" != "x"],[
+    CXXFLAGS="$LIBLDAP_CFLAGS $CXXFLAGS"
```

See https://github.com/squid-cache/squid/pull/1153 and related PKG_CHECK_MODULES() [documentation](https://autotools.info/pkgconfig/pkg_check_modules.html).

```suggestion
    CPPFLAGS="$LIBLDAP_CFLAGS $CPPFLAGS"
```